lib/distributions/Circle/FIGDistribution.m

```diff
@@ -0,0 +1,291 @@
+            %       n-th trigonometric moment (complex number)
+            noCoeffs = numel(this.gridValues);
+            switch this.transformation
+                case 'identity'
+                    fvals = this.gridValues;
+                case 'sqrt'
+                    fvals = this.gridValues.^2;
+                otherwise
+                    error('Moment:unsupportedTransformation', 'Transformation not recognized or unsupported');
+            end
+            if mod(numel(this.gridValues), 2) == 0
+                warnStruct = warning('off', 'Truncate:TooFewCoefficients');
+                noCoeffs = noCoeffs + 1;
+            end
+            fd = FourierDistribution.fromFunctionValues(fvals', noCoeffs, 'identity');
+            if mod(numel(this.gridValues), 2) == 0
+                warning(warnStruct);
+            end
+            m = fd.trigonometricMoment(n);
+        end
+        
+        function p = plot(this, varargin)
+            gridPoints = 0:2 * pi / numel(this.gridValues):2 * pi - 2 * pi / numel(this.gridValues);
+            holdStatus = ishold;
+            p(1) = plot@AbstractCircularDistribution(this, varargin{:});
+            hold on
+            switch this.transformation
+                case 'identity'
+                    p(2) = plot(gridPoints, this.gridValues, 'x');
+                case 'sqrt'
+                    p(2) = plot(gridPoints, this.gridValues.^2, 'x');
+                otherwise
+                    error('Transformation unsupported');
+            end
+            if ~holdStatus
+                hold off
+            end
+        end
+        
+        function p = value(this, xa)
+            % Value when interpreted as a pmf
+            gridPoints = (0:2 * pi / numel(this.gridValues):2 * pi - 2 * pi / numel(this.gridValues))';
+            p = (repmat(xa, 1, 5) == gridPoints)' * this.gridValues;
+        end
+        
+        function f = multiply(this, f2)
+            % Multiplies two transformed fourier pdfs (returns transformed result)
+            if ~strcmp(this.transformation, f2.transformation)
+                error('Multiply:differentTransformations', 'Transformations do not match, transform before using multiply');
+            end
+            if ~(strcmp(this.transformation, 'identity') || strcmp(this.transformation, 'sqrt'))
+                error('Multiply:unsupportedTransformation', 'Transformation not recognized or unsupported');
+            end
+            assert(this.enforcePdfNonnegative == f2.enforcePdfNonnegative);
+            assert(numel(this.gridValues) == numel(f2.gridValues));
+            warnStruct = warning('off', 'Normalization:notNormalized');
+            f = FIGDistribution(this.gridValues.*f2.gridValues, this.transformation, this.enforcePdfNonnegative);
+            warning(warnStruct);
+        end
+        
+        function f = convolve(this, f2)
+            % Multiplies two transformed fourier pdfs (returns transformed result)
```

Comment:
You may want to double check the doc string here. The function says `convolve` while the text says `multiply`.